SquareLine/gen.py

```diff
@@ -0,0 +1,231 @@
+# Generated output directory
+OUT_DIR = "espressif"
+# SquareLine output directory
+SQUARELINE_OUTDIR = "__ui_project_name__"
+# Template of SLB file
+SLB_FILE = {
+    "version":"",
+    "group":"Espressif",
+    "title": "",
+    "keywords": "Espressif",
+    "width": 0,
+    "height": 0,
+    "width_min": 0,
+    "height_min": 0,
+    "width_max": 0,
+    "height_max": 0,
+    "offset_x": 0,
+    "offset_y": 0,
+    "rotation": 0,
+    "color_depth": "",
+    "lvgl_export_path": "",
+    "lvgl_include_path": "lvgl.h",
+    "supported_lvgl_version": "8.0+",
+    "pattern_match_files": "./CMakeLists.txt",
+    "language":"C",
+    "ui_export_path":"./main/ui",
+    "url":"https://github.com/espressif/esp-bsp",
+    "short_description": "",
+    "long_description": ""
+}
+
+global placeholders
+
+# Print green text
+def print_ok(str):
+    print(Fore.GREEN + str + Fore.WHITE)
+
+# Print red error
+def print_error(str):
+    print(Fore.RED + str + Fore.WHITE)
+
+def remove_folder(path):
+    for filename in os.listdir(path):
+        f = os.path.join(path, filename)
+        if os.path.isdir(f):
+            remove_folder(path + "/" + filename)
+        else:
+            os.remove(path + "/" + filename)
+    os.rmdir(path)
+
+class SafeDict(dict):
+    def __missing__(self, key):
+        return '{' + key + '}'
+
+def replace_placeholders(file):
+    global placeholders
+    if os.path.exists(file):
+        f = open(file, "r")
+        str = f.read()
+        f.close()
+        str = str.format_map(SafeDict(placeholders))
+        f = open(file, "w")
+        f.write(str)
+        f.close()
+
+# Copy file and replace placeholders in copied file
+def copy_file(src_path, dst_path, translate=1):
+    print(f"  Copying {src_path} to {dst_path}")
+    shutil.copyfile(src_path, dst_path)
+    if translate:
+        replace_placeholders(dst_path)
+
+# Copy all files in directory
+def copy_directory(from_dir, to_dir):
+    if not os.path.exists(to_dir):
+        os.makedirs(to_dir)
+    for filename in os.listdir(from_dir):
+        f = os.path.join(from_dir, filename)
+        out_f = os.path.join(to_dir, filename)
+        if os.path.isdir(f):
+            copy_directory(f, out_f)
+        else:
+            copy_file(f, out_f)
+
+def zip_add_folder(zip_obj, path, outdir):
+    for filename in os.listdir(path):
+        file = os.path.join(path, filename)
+        zip_obj.write(file, outdir + "/" + os.path.basename(file))
+        if os.path.isdir(file):
+            zip_add_folder(zip_obj, file, outdir + "/" + filename)
+
+def make_zip(board_name, path):
+    zip_obj = ZipFile(path + "/" + board_name + ".zip", 'w')
+    zip_dir = path + "/" + SQUARELINE_OUTDIR
+    zip_add_folder(zip_obj, zip_dir, SQUARELINE_OUTDIR)
+    zip_obj.close()
+
+def check_json_key(manifest, key):
+    if key not in manifest:
+        print_error(f"Missing {key} in manifest JSON file")
+        os._exit(1)
```

Comment:
```suggestion
        raise SystemExit(1)
```

(same for the other cases)

Reply:
or `sys.exit(1)` if you don't print a backtrace.